### DISCUSSION TITLE
runtimeConfig doesn't see environment variables in Nuxt 3

### DISCUSSION THREAD

**Discussion by aslanovshm** [nuxt/nuxt#17200] [Questions]
Hello

**Steps to reproduce:**
1. I do `yarn build` without the `.env` file
2. Add the `.env` file to the project
3. I do `yarn start`. 
4. Print the `useRuntimeConfig().public.baseURL` to the console and get `undefined`. 

Why is runtime not tracking my environment variables?

<img width="494" alt="image" src="https://user-images.githubusercontent.com/36332769/180813920-ba7a98bd-0593-4234-a289-161228dff439.png">

**.env**
```
NUXT_PUBLIC_BASE_URL=https://example.com/api/v1
```
**nuxt.config.js**
```ts
export default defineNuxtConfig({
  runtimeConfig: {
    public: {
      baseUrl: ''
    }
  },
```
**plugins/app.js**
```js
export default defineNuxtPlugin(() => {
  console.log('baseURL', useRuntimeConfig().public.baseURL
})
```

**Versions:**
Node.js: 16.13.0
NPM: 8.1.0
Yarn: 1.22.17
Nuxt: 3.0.0-rc.4

**Comment by peterjunsworth** [comment]
```
runtimeConfig: {
    public: {
      baseUrl: process.env.NUXT_PUBLIC_BASE_URL
    }
  }
```

**Reply by aslanovshm** [reply]
Thanks for your answer, i tried to do the same, but unfortunely it doesn't work for me

**Reply by peterjunsworth** [reply]
Make sure you restart your server with config changes. Also add quotes in your .env:
NUXT_PUBLIC_BASE_URL="https://example.com/api/v1"

Try logging out the below:
```
const config = useRuntimeConfig();
console.log(config.public.baseUrl);
```

**Reply by danielroe** [reply]
Note that you should define a fallback value in your config file. If the value is undefined when building it will not allow runtime overrides.

**Reply by peterjunsworth** [reply]
Funnily enough - I came across same issue just now when deploying to remote server. Locally works fine but on remote, plugin files can't read the runtimeConfig. runtimeConfig works fine in non-plugin files but I had to point my plugin file directly to my .env var

https://github.com/nuxt/bridge/issues/9
